src/openassetio-python/tests/package/hostApi/test_manager.py

```diff
@@ -258,6 +259,397 @@ def test_when_entity_ref_prefix_type_invalid_then_debug_log_printed(
 )
 
 
+class BatchFirstMethodTest:
+    subtests: Any
+    method: Callable
+    invoke_success_cb: Callable
+    invoke_error_cb: Callable
+    mock_interface_method: mock.Mock
+
+    a_batch_element_error: BatchElementError
+    a_context: Context
+    a_host_session: managerApi.HostSession
+
+    batch_element_error_codes = list(BatchElementError.ErrorCode.__members__.values())
+
+    batch_element_error_codes_names = [
+        "unknown",
+        "invalidEntityReference",
+        "malformedEntityReference",
+        "entityAccessError",
+        "entityResolutionError",
+        "invalidPreflightHint",
+        "invalidTraitSet",
+    ]
+
+    def assert_callback_overload_wraps_the_corresponding_method_of_the_held_interface(
+        self, method_specific_args_for_batch_of_two, one_success_result
+    ):
+        success_callback = mock.Mock()
+        error_callback = mock.Mock()
+
+        def call_callbacks(*_args):
+            self.invoke_success_cb(123, one_success_result)
+            self.invoke_error_cb(456, self.a_batch_element_error)
+
+        self.mock_interface_method.side_effect = call_callbacks
+
+        self.method(
+            *method_specific_args_for_batch_of_two,
+            self.a_context,
+            success_callback,
+            error_callback,
+        )
+
+        self.mock_interface_method.assert_called_once_with(
+            *method_specific_args_for_batch_of_two,
+            self.a_context,
+            self.a_host_session,
+            mock.ANY,
+            mock.ANY,
+        )
+
+        success_callback.assert_called_once_with(123, one_success_result)
+        error_callback.assert_called_once_with(456, self.a_batch_element_error)
+
+    def assert_callback_overload_with_mixed_array_lengths(
```

Comment:
[Consider] This and others.
```suggestion
    def assert_callback_overload_errors_with_mixed_array_lengths(
```

Reply:
Updated in 37436e8d (just two methods that I could see, any others?)